File: app/src/main/java/com/example/projeto_ds2/model/mensagem/Mensagem.java
package com.example.projeto_ds2.model.mensagem;

import java.io.Serializable;

public class Mensagem implements Serializable {

    private static final long serialVersionUID = -4740438769141831046L;

    private OrigemEnum origem;
    private String texto;

    public Mensagem(OrigemEnum origem, String texto, String hora) {
        this.origem = origem;
        this.texto = texto;
        this.hora = hora;
    }

    public String getHora() {
        return hora;
    }

    public void setHora(String hora) {
        this.hora = hora;
    }

    private String hora;

    public Mensagem(OrigemEnum origem, String texto) {
        this.origem = origem;
        this.texto = texto;
    }

    public OrigemEnum getOrigem() {
        return origem;
    }

    public void setOrigem(OrigemEnum origem) {
        this.origem = origem;
    }

    public String getTexto() {
        return texto;
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }
}

File: app/src/main/java/br/edu/up/vivianmarcal/model/mensagem/Mensagem.kt
package br.edu.up.vivianmarcal.model.mensagem

import br.edu.up.vivianmarcal.firebase.FirebaseConstants
import br.edu.up.vivianmarcal.model.usuario.Usuario
import com.google.firebase.Timestamp
import java.io.Serializable
import java.util.ArrayList
import kotlin.collections.HashMap

class Mensagem : Serializable {
    var usuario: Usuario? = null
    var origem: OrigemEnum? = null
    private var nome: String? = null
    var texto: String
    var data: Timestamp? = null

    constructor(origem: OrigemEnum?, texto: String, data: Timestamp?) {
        this.origem = origem
        this.texto = texto
        this.data = data
    }

    constructor(usuario: Usuario?, texto: String) {
        this.usuario = usuario
        this.texto = texto
    }


    fun getHash(): HashMap<String, Any?> {
        return hashMapOf(
            FirebaseConstants.MENSAGENS_FIELD_USUARIO to usuario!!.getHash(),
            FirebaseConstants.MENSAGENS_FIELD_DATA to Timestamp.now(),
            FirebaseConstants.MENSAGENS_FIELD_CORPO to texto
        )
    }

    companion object {
        private const val serialVersionUID = -4740438769141831046L
        val instanceList: ArrayList<Mensagem>
            get() = ArrayList()
    }
}